### PR TITLE
[doc] Contribute a new shape to specify aspect ratio on node

### LINKED ISSUE #2255: Let nodes keep a specific aspect ratio
- It should be possible to define nodes which should stick to a specific aspect ratio (for example, I want an image based node to stick to a `1:1` or `16:9` aspect ratio.
- See `NodeResizer#keepAspectRatio` and the layout algorithm to enforce the aspect ratio on the frontend
- Unit tests should be added to ensure that the layout algorithm keeps the aspect ratio in all conditions

### PR DESCRIPTION
**Pull request by frouene**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [x] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/iterations/2023.12/shapes/keep_node_aspect_ratio.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+There is no possibility to configure a node to maintain its ratio in every case.
+
+== Key Result
+
+A specifier should be able to fix the default size on a node and specify if this ratio should be fixed.
+The diagram representation should apply these default values and keep the ratio in every layout situation.
+
+== Solution
+
+Add to the diagram DSL for `NodeDescription`:
```
Comment (pcdavid) [inline doc/iterations/2023.12/shapes/keep_node_aspect_ratio.adoc:14]: Shoudldn't these be on `NodeStyleDescription`?
In the presence of conditional style, I see no particular reason to assume that each style variant would have the same default size and `keepAspectRatio` value.

But then this opens the question: how are the new `default{Width,Height}Expression` different from the existing `{width,height}ComputationExpression` and how do they interact if we have both?

Comment (frouene) [reply doc/iterations/2023.12/shapes/keep_node_aspect_ratio.adoc:14]: After discussion with @sbegaudeau, we leave only the size properties in the NodeDescription and removed the old one `{width,height}ComputationExpression`

File: doc/iterations/2023.12/shapes/keep_node_aspect_ratio.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+* `defaultWitdhExpression`, the expression to retrieve the default width to apply to the node.
+* `defaultHeightExpression`, the expression to retrieve the default height to apply to the node.
+* `keepAspectRatio`, if true the ratio width/height should always be conserved.
```
Comment (pcdavid) [inline doc/iterations/2023.12/shapes/keep_node_aspect_ratio.adoc]: "conserved" => "preserved"

**Review by pcdavid — APPROVED** [review]
I'll fix the small wording issue and rebase/merge.

**Pull request merged**